### PR TITLE
CI: allow the cirrus-ci run to be skipped

### LINKED ISSUE #17252: ENH: CI: Add a `[skip cirrus]` markup to skip running Cirrus jobs.
It would be nice to be able to skip the Cirrus jobs, especially in their current state.  It looks like it might be as simple as adding something like 

```
    if "[skip cirrus]" in dct["message"]:
        return []
```

at the appropriate place in [`.cirrus.star`](https://github.com/scipy/scipy/blob/main/.cirrus.star), but that idea is the result of about 5 minutes of code browsing.  @andyfaff, would something like that work?

### PR DESCRIPTION
**Pull request by andyfaff**
closes #17252

### CODE REVIEW

**Comment by andyfaff** [comment]
@WarrenWeckesser

File: .cirrus.star [hunk 1]
```diff
@@ -35,4 +35,7 @@ def main(ctx):
     # this configuration runs a single linux_aarch64 + macosx_arm64 run.
     # there's no need to do this during a wheel run as they automatically build
     # and test over a wider range of Pythons.
+    if "[skip cirrus]" in dct["message"]:
```
Comment (WarrenWeckesser) [inline .cirrus.star:38]: This also needs to check for `[skip ci]`.  (Sorry, I didn't think of this until now.)

Comment (andyfaff) [reply .cirrus.star]: done

**Review by WarrenWeckesser — APPROVED** [review]
Looks good to me (not that I'm a Cirrus starlark expert). Let's try it.

**Comment by WarrenWeckesser** [comment]
If I squash and merge this, should I preserve any or all of the `skip` annotations in the squashed commit?

**Comment by andyfaff** [comment]
I don't think it matters whether you keep the commit messages or not

**Comment by WarrenWeckesser** [comment]
OK, merged!

**Pull request merged**